server.go

```diff
@@ -183,74 +201,45 @@ func (s *SockchatServer) editProfile(w http.ResponseWriter, r *http.Request) {
-		return
-	}
-	channel, _ := s.store.GetChannel(req.Channel)
-	for user := range channel.Users {
-		user.WriteSocketMsg(NewSocketMessage("new_message", MessageEvent{Channel: req.Channel, Author: conn.nick, Text: req.Text}))
+	if conn.authorized {
+		if handler, exists := s.authorizedUsers.GetHandler(conn.Nick); exists {
+			return handler, nil
+		}
+		log.Print("warning: received a connection which was authorized but handler is not found")
 	}
 
-}
-
-func (s *SockchatServer) loginUser(ctx context.Context, request SocketMessage, conn *SockChatWS) {
+	if request.Action != LoginAction {
+		return nil, fmt.Errorf("you must log in first using " + LoginAction + " action")
+	}
 	req := UnmarshalLoginRequest(request.Payload)
-	if err := s.userService.LoginUser(ctx, req.Nick, req.Password); err != nil {
-		conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-	} else {
-		conn.nick = req.Nick
-		conn.WriteSocketMsg(NewSocketMessage(fmt.Sprintf("logged_in:%s", req.Nick), "{}"))
-		conn.SetReadDeadline(time.Now().Add(s.timeoutAuthorized))
+	if !s.userProfiles.IsAuthValid(ctx, req.Nick, req.Password) {
+		return nil, fmt.Errorf("login rejected: invalid credentials")
 	}
-}
 
-func (s *SockchatServer) createNewChannel(request SocketMessage, conn *SockChatWS) {
-	channel := UnmarshalChannelRequest(request.Payload)
-	if err := s.store.CreateChannel(channel.Name); err != nil {
-		conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-	} else {
-		conn.WriteSocketMsg(NewSocketMessage("channel_created", channel))
-	}
-}
+	conn.Nick = req.Nick
+	conn.authorized = true
 
-func (s *SockchatServer) joinChannel(request SocketMessage, conn *SockChatWS) {
-	channel := UnmarshalChannelRequest(request.Payload)
-	if err := s.store.AddUserToChannel(channel.Name, conn); err != nil {
-		conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-	} else {
-		conn.WriteSocketMsg(NewSocketMessage("channel_joined", ChannelUserChangeEvent{Channel: channel.Name, Nick: conn.nick}))
-	}
+	handler := s.authorizedUsers.AddConnection(conn, s.channelStore)
+	conn.WriteSocketMsg(NewSocketMessage("logged_in:"+req.Nick, "{}"))
+	return handler, nil
 }
 
-func (s *SockchatServer) leaveChannel(request SocketMessage, conn *SockChatWS) {
-	channel := ChannelRequest{}
-	if err := json.Unmarshal(request.Payload, &channel); err != nil {
-		log.Printf("error while unmarshaling request for leaving channel: %v", err)
-	}
-	if err := s.store.RemoveUserFromChannel(channel.Name, conn); err != nil {
-		conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-	} else {
-		conn.WriteSocketMsg(NewSocketMessage("channel_left", ChannelUserChangeEvent{Channel: channel.Name, Nick: conn.nick}))
+func (s *SockchatServer) shutConnection(conn *SockChatWS) {
+	if conn.authorized {
+		s.authorizedUsers.RemoveConnection(conn)
 	}
+	conn.Close()
 }
 
 type SockChatWS struct {
 	*websocket.Conn
-	writeLock sync.Mutex
-	readLock  sync.Mutex
-	nick      string
+	writeLock  sync.Mutex
+	readLock   sync.Mutex
+	Nick       string
```

Comment:
```suggestion
	nick       string
```
Nick field doesn't have to be exported